src/Microsoft.Data/ArrowStringColumn.cs

```diff
@@ -0,0 +1,348 @@
+                mutableDataBuffer.Length += bytes.Length;
+                mutableOffsetsBuffer.Append(mutableOffsetsBuffer[mutableOffsetsBuffer.Length - 1] + bytes.Length);
+            }
+            SetValidityBit(Length - 1, value == null ? true : false);
+        }
+
+        private int GetBufferIndexContainingRowIndex(ref long rowIndex)
+        {
+            if (rowIndex >= Length)
+            {
+                throw new ArgumentOutOfRangeException(Strings.ColumnIndexOutOfRange, nameof(rowIndex));
+            }
+
+            // Since the strings here could be of variable length, scan linearly
+            int curArrayIndex = 0;
+            int numBuffers = _offsetsBuffers.Count;
+            while (curArrayIndex < numBuffers && rowIndex > _offsetsBuffers[curArrayIndex].Length - 1)
+            {
+                rowIndex -= _offsetsBuffers[curArrayIndex].Length - 1;
+                curArrayIndex++;
+            }
+            return curArrayIndex;
+        }
+
+        private ReadOnlySpan<byte> GetBytes(long index)
+        {
+            int offsetsBufferIndex = GetBufferIndexContainingRowIndex(ref index);
+            int currentOffset = _offsetsBuffers[offsetsBufferIndex].ReadOnlySpan[(int)index];
+            int nextOffset = _offsetsBuffers[offsetsBufferIndex].ReadOnlySpan[(int)index + 1];
+            int numberOfBytes = nextOffset - currentOffset;
+            return _dataBuffers[offsetsBufferIndex].ReadOnlySpan.Slice(currentOffset, numberOfBytes);
+        }
+
+        protected override object GetValue(long rowIndex)
+        {
+            var bytes = GetBytes(rowIndex);
+            unsafe
+            {
+                fixed (byte* data = &MemoryMarshal.GetReference(bytes))
+                    return _encoding.GetString(data, bytes.Length);
+            }
+        }
+
+        protected override object GetValue(long startIndex, int length)
+        {
+            var ret = new List<string>();
+            while (ret.Count < length)
+            {
+                ret.Add((string)GetValue((int)startIndex));
+            }
+            return ret;
+        }
+
+        protected override void SetValue(long rowIndex, object value)
+        {
+            throw new NotSupportedException(Strings.ImmutableStringColumn);
+        }
+
+        public new string this[long rowIndex]
+        {
+            get => (string)GetValue(rowIndex);
+            set => throw new NotSupportedException(Strings.ImmutableStringColumn);
+        }
+
+        public new List<string> this[long startIndex, int length]
+        {
+            get
+            {
+                var ret = new List<string>();
+                while (ret.Count < length)
+                {
+                    ret.Add((string)GetValue(startIndex++));
+                }
+                return ret;
+            }
+        }
+
+        public override Field Field => new Field(Name, StringType.Default, NullCount != 0);
```

Comment:
In general, this isn't a good pattern. A property should act more or less like a field, it should be quick to access it. Here you are creating a new object every time the property is accessed. If this is intended, it should be a method.